### ISSUE TITLE
VPN authentication error with nordvpn/openvpn?

### ISSUE THREAD

**Issue by SherifG97** [DyonR/docker-qbittorrentvpn#182]
I am tryign to setupt he docker on my windows machine and I keep getting this error no matter what I try:

`2023-05-15 11:38:26 2023-05-15 15:38:26 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
2023-05-15 11:38:26 2023-05-15 15:38:26 VERIFY OK: depth=1, O=NordVPN, CN=NordVPN CA8
2023-05-15 11:38:26 2023-05-15 15:38:26 VERIFY KU OK
2023-05-15 11:38:26 2023-05-15 15:38:26 Validating certificate extended key usage
2023-05-15 11:38:26 2023-05-15 15:38:26 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2023-05-15 11:38:26 2023-05-15 15:38:26 VERIFY EKU OK
2023-05-15 11:38:26 2023-05-15 15:38:26 VERIFY X509NAME OK: CN=ca1638.nordvpn.com
2023-05-15 11:38:26 2023-05-15 15:38:26 VERIFY OK: depth=0, CN=ca1638.nordvpn.com
2023-05-15 11:38:26 2023-05-15 15:38:26 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 4096 bit RSA
2023-05-15 11:38:26 2023-05-15 15:38:26 [ca1638.nordvpn.com] Peer Connection Initiated with [AF_INET]45.88.190.130:443
2023-05-15 11:38:27 2023-05-15 15:38:27 SENT CONTROL [ca1638.nordvpn.com]: 'PUSH_REQUEST' (status=1)
2023-05-15 11:38:33 2023-05-15 15:38:33 SENT CONTROL [ca1638.nordvpn.com]: 'PUSH_REQUEST' (status=1)
2023-05-15 11:38:33 2023-05-15 15:38:33 AUTH: Received control message: AUTH_FAILED
2023-05-15 11:38:33 2023-05-15 15:38:33 SIGTERM[soft,auth-failure] received, process exiting`

I am using these docker settings:
docker run --privileged -d -v "C:\docker config\qbittorrentvpn\config":/config -v "M:\Downloads\torrentqbit"/:/downloads -e "VPN_ENABLED=yes" -e "VPN_TYPE=openvpn" -e "LAN_NETWORK=192.168.2.1/32" -e "VPN_USERNAME=xxxx" -e "VPN_PASSWORD=yyyy" -p 8083:8080 --restart unless-stopped dyonr/qbittorrentvpn

I have checked my username and password are correct. The OpenVPN config file is in the correct location. Am I missing something obvious here?

**Comment by wpkato** [comment]
Just started up an instance and I'm receiving the same error when using NordVPN

**Comment by vnnegt** [comment]
> Just started up an instance and I'm receiving the same error when using NordVPN

post your docker command, file, etc please

**Comment by wpkato** [comment]
nm, I figured it out with some google-fu

In my case anyways, NordVPN this month (June, 2023), changed the way how they allow 3rd party apps access to their services.  You can't use your straight userid/passwd.  You have to get credentials from within your account dashboard (which worked for me) or use a token (which I didn't bother with since the first method worked for me.)  YMMV.  Hope this helps.

**Comment by CanadianBaconPro** [comment]
> nm, I figured it out with some google-fu
> 
> In my case anyways, NordVPN this month (June, 2023), changed the way how they allow 3rd party apps access to their services. You can't use your straight userid/passwd. You have to get credentials from within your account dashboard (which worked for me) or use a token (which I didn't bother with since the first method worked for me.) YMMV. Hope this helps.

Thank you! I've been banging my head against a wall as everything in my setup was working, until it wasn't. I hadn't changed anything and kept receiving auth failed. This is 100% a Nord issue and using the service credentials fixed it for me as well.

**Comment by noviceuser** [comment]
I tried changing my credentials.conf to the hashed user and pass from NordVPN site but I still get the same issue.
Or are you placing the hased U/P in the docker-compose.yml?
Does anyone have instructions od doing this? (in case I messed it up.) or even better anyone have instructions of how to use a token? I can get a token OK. just don't know where to put it. and what setting to change in docker/qbittorrent

**Comment by DyonR** [comment]
> I tried changing my credentials.conf to the hashed user and pass from NordVPN site but I still get the same issue. Or are you placing the hased U/P in the docker-compose.yml? Does anyone have instructions od doing this? (in case I messed it up.) or even better anyone have instructions of how to use a token? I can get a token OK. just don't know where to put it. and what setting to change in docker/qbittorrent  

Get your NordVPN credentials here: https://my.nordaccount.com/dashboard/nordvpn/  
If you need something more visual, this guide is for Android, but the process of obtaining the NordPVN credentials are the same: https://support.nordvpn.com/Connectivity/Android/1114905392/Manual-OpenVPN-connection-setup-for-Android.htm  
Just skip to the section which starts with `"You can find your NordVPN service credentials"`. Any other instructions you may ignore. All you need is just the VPN credentials.
  
Just set those in your environment variables

**Comment by noviceuser** [comment]
> > I tried changing my credentials.conf to the hashed user and pass from NordVPN site but I still get the same issue. Or are you placing the hased U/P in the docker-compose.yml? Does anyone have instructions od doing this? (in case I messed it up.) or even better anyone have instructions of how to use a token? I can get a token OK. just don't know where to put it. and what setting to change in docker/qbittorrent
> 
> Get your NordVPN credentials here: https://my.nordaccount.com/dashboard/nordvpn/ If you need something more visual, this guide is for Android, but the process of obtaining the NordPVN credentials are the same: https://support.nordvpn.com/Connectivity/Android/1114905392/Manual-OpenVPN-connection-setup-for-Android.htm Just skip to the section which starts with `"You can find your NordVPN service credentials"`. Any other instructions you may ignore. All you need is just the VPN credentials.
> 
> Just set those in your environment variables

Thanks for the reply. So my process should be:
1. Edit *.ovpn file and remove auth-user-pass credentials.conf (as we are not longer using credentials.conf)
2. add user and pass as environment variables to my docker compose.
3. Restart container

I dont want the entry in the .ovpn to override or messup environmental variables. havn't had to touch this container in ages so I want it to be neet and tidy when I go back to it again in 6 months.

**Comment by DyonR** [comment]
@noviceuser  No, you still use the environment variables of this container with Username and Password. 
NordVPN only changed the credentials. Instead of using login credentials for the website for .ovpn files, they now use separate credentials for .ovpn files. 
credentials.conf is made by the container. You don't make this yourself. It uses the USERNAME and PASSWORD variable from the environment variables. 
All you do is replace the USERNAME and PASSWORD with the new credentials from NordVPN

**Comment by noviceuser** [comment]
@DyonR That worked. Thanks for your patience. I learnt a little bit more today. cheers
